crates/network-primitives/src/traits.rs

```diff
@@ -107,6 +124,25 @@ impl<T: TransactionResponse> TransactionResponse for WithOtherFields<T> {
     fn input(&self) -> &Bytes {
         self.inner.input()
     }
+
+    fn gas_price(tx: &impl alloy_consensus::Transaction, base_fee: Option<u64>) -> u128 {
+        T::gas_price(tx, base_fee)
+    }
+
+    fn max_fee_per_gas(tx: impl alloy_consensus::Transaction) -> Option<u128> {
+        T::max_fee_per_gas(tx)
+    }
```

Comment:
these don't make much sense because the caller can just call them on tx directly

Reply:
https://github.com/alloy-rs/alloy/pull/1173#discussion_r1732734183